journal/src/test/java/io/zeebe/journal/file/SegmentedJournalReaderTest.java

```diff
@@ -78,4 +80,164 @@ public void shouldReadAfterCompact() {
     assertThat(reader.hasNext()).isTrue();
     assertThat(reader.next().index()).isEqualTo(indexToCompact);
   }
+
+  @Test
```

Comment:
Nit: Looks like the following tests belong to JournalReaderTest because they are testing the API.

Reply:
Hm, it's testing with multiple segments though which I guess we wouldn't do in the API itself? What's the current approach for this, how did we decide to differentiate these things?

Reply:
My idea is - if it has to test something specific to segements, it should go to Segmented*Test. For eg:- `shouldSeekToLast` should be in API test, but `shouldSeekToLastWhenMoreThanOneSegment` would go to Segmented*Test.